cohttp-eio/tests/client.md

```diff
@@ -0,0 +1,233 @@
+## Setup
+
+```ocaml
+open Eio.Std
+open Cohttp_eio
+```
+
+A mock client socket and host for testing:
+
+```ocaml
+let host = ("localhost", None)
+let conn = Eio_mock.Flow.make "socket"
+let run ~response ~test =
+  Eio_mock.Backend.run @@ fun () ->
+  Fiber.both
+    (fun () -> Eio_mock.Flow.on_read conn response)
+    test
+```
+
+## Tests
+
+GET method request:
+
+```ocaml
+# run ~response:
+    [`Return "HTTP/1.1 200 OK\r\n";
+     `Return "content-length: 4\r\n";
+     `Return "content-type: text/plain; charset=UTF-8\r\n\r\n";
+     `Return "root";
+     `Raise End_of_file
+    ]
+    ~test:(fun () ->
+      (Client.get
+        ~headers:(Http.Header.of_list [ ("Accept", "application/json") ])
+        ~conn
+        host
+        "/")
+      |> Client.read_fixed
+      |> ignore);;
```

Comment:
Why are you ignoring the result? This isn't testing anything except that the client read from the socket!

(this comment applies to the other tests in this file too)

Reply:
> Why are you ignoring the result? This isn't testing anything except that the client read from the socket!

The tests in client.md mostly tests the `Cohttp_eio.Client` response headers and that we can successfully receive response without any exceptions/errors. However, yes it does seem wasteful to ignore the `read_fixed` results. I have modified the tests to check for valid response body where possible.